### ISSUE TITLE
Group operation chaining produces unexpected result with group all

### ISSUE THREAD

**Issue by affo** [influxdata/flux#322]
If we chain multiple `group` operations, we get the expected result, so that the last `group` operation is the only one that matters.

I.e., this:

```
from(bucket: "test")
    |> range(start: -1y)
    |> group(none: true)
    |> group(by: ["tag1"])
    |> group(all: true)
    |> ...
    |> group(by: ["tag1", "tag2"])
```

is equivalent to:

```
from(bucket: "test")
    |> range(start: -1y)
    |> group(by: ["tag1", "tag2"])
```

However, if the last group is `group(all: true)`, it has the same effect as a `group(none: true)`, we get only one table with an empty group key.  

In other words, this:

```
from(bucket: "test")
    |> range(start: -1y)
    |> group(by: ["tag1"])
    |> group(all: true)
```

Is _not_ equivalent to:

```
from(bucket: "test")
    |> range(start: -1y)
    |> group(all: true)
```

And produces a single table with no group key.

**Comment by affo** [comment]
Should we implement a rewrite rule that keeps only the last `group` operation?

**Comment by affo** [comment]
@stuartcarnie can you comment on this?

**Comment by stuartcarnie** [comment]
@affo I suspect a planner or execution bug, however, I agree, we should be able to rewrite multiple _consecutive_ `group` operations to drop all but the last one. @wolffcm does that sounds correct to you?

**Comment by wolffcm** [comment]
@stuartcarnie Yes, that sounds right to me.

**Comment by wolffcm** [comment]
I'll have a look at this one.

**Comment by affo** [comment]
Yep, I noticed that in `Process`, if the mode is `All`, no group key is composed: https://github.com/influxdata/flux/blob/master/functions/transformations/group.go#L203.

Anyhow, just to add some more issues, the `group()` thing is something we talked with @nathanielc and could be considered bad behavior: #316.

**Comment by affo** [comment]
@wolffcm I'd love to review this once you are done.

**Comment by wolffcm** [comment]
@affo I made you a reviewer on the pull request.

I also filed an issue for the rewrite rule:
https://github.com/influxdata/flux/issues/339
